src/NServiceBus.Core/Transports/MessageContext.cs

```diff
@@ -11,30 +11,30 @@ public partial class MessageContext : IExtendable
         /// <summary>
         /// Initializes the context.
         /// </summary>
-        /// <param name="messageId">Native message id.</param>
+        /// <param name="transportMessageId">Native message id.</param>
         /// <param name="headers">The message headers.</param>
         /// <param name="body">The message body.</param>
         /// <param name="transportTransaction">Transaction (along with connection if applicable) used to receive the message.</param>
         /// <param name="context">A <see cref="ContextBag" /> which can be used to extend the current object.</param>
-        public MessageContext(string messageId, Dictionary<string, string> headers, byte[] body, TransportTransaction transportTransaction, ContextBag context)
+        public MessageContext(string transportMessageId, Dictionary<string, string> headers, byte[] body, TransportTransaction transportTransaction, ContextBag context)
         {
-            Guard.AgainstNullAndEmpty(nameof(messageId), messageId);
+            Guard.AgainstNullAndEmpty(nameof(transportMessageId), transportMessageId);
             Guard.AgainstNull(nameof(body), body);
             Guard.AgainstNull(nameof(headers), headers);
             Guard.AgainstNull(nameof(transportTransaction), transportTransaction);
             Guard.AgainstNull(nameof(context), context);
 
             Headers = headers;
             Body = body;
-            MessageId = messageId;
+            TransportMessageId = transportMessageId;
             Extensions = context;
             TransportTransaction = transportTransaction;
         }
 
         /// <summary>
         /// The native id of the message.
         /// </summary>
-        public string MessageId { get; }
+        public string TransportMessageId { get; }
```

Comment:
Taking stock of the various ID's:

```c#
MessageContext.TransportMessageId

ErrorContext.Message.MessageId
ErrorContext.Message.NativeMessageId

CompleteContext.TransportMessageId
```

Something seems off with this. Does any ID flow through all three contexts? If so, I would expect it to be named the same in each.

Reply:
Will rename to NativeMessageId